docs/source/guides/statistical_insights.ipynb

```diff
@@ -209,6 +209,209 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.DataFrame({\n",
+    "    \"idx\": range(100),\n",
+    "    \"dt1\": pd.date_range(\"2005-01-01\", periods=100, freq=\"H\"),\n",
+    "    \"dt2\": pd.date_range(\"2005-01-01\", periods=100, freq=\"B\")\n",
+    "})\n",
+    "df.ww.init()\n",
+    "df.ww.infer_temporal_frequencies()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "{\n",
+    "    'dt1': 'H', \n",
+    "    'dt2': 'B'\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Inferring Noisy data (Missing Values)\n",
+    "If your timeseries is noisy, and you pass a `debug=True` flag, the returned dictionary will also include debug objects for each temporal column. This object is useful in helping you understand where in your data there is a problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dt1_a = pd.date_range(end=\"2005-01-01 10:00:00\", periods=500, freq=\"H\")\n",
+    "dt1_b = pd.date_range(start=\"2005-01-01 15:00:00\", periods=500, freq=\"H\")\n",
+    "\n",
+    "\n",
+    "df = pd.DataFrame({\n",
+    "    \"idx\": range(1000),\n",
+    "    \"dt1\": dt1_a.append(dt1_b),\n",
+    "})\n",
+    "df.ww.init()\n",
+    "df.ww.infer_temporal_frequencies(debug=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "{\n",
+    "    'dt1': (\n",
+    "        None, {\n",
+    "            'actual_range_start': '2004-12-11T15:00:00',\n",
+    "            'actual_range_end': '2005-01-22T10:00:00',\n",
+    "            'message': None,\n",
+    "            'estimated_freq': 'H',\n",
+    "            'estimated_range_start': '2004-12-11T15:00:00',\n",
+    "            'estimated_range_end': '2005-01-22T10:00:00',\n",
+    "            'duplicate_values': [],\n",
+    "            'missing_values': [\n",
+    "                {'dt': '2005-01-01T11:00:00', 'idx': 500, 'range': 4}\n",
+    "            ],\n",
+    "            'extra_values': [],\n",
+    "            'nan_values': []\n",
+    "        }\n",
+    "    )\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Debug Object Description\n",
+    "\n",
+    "The debug object contains the following information:\n",
+    "\n",
+    "- **actual_range_start**: an ISO 8601 formatted string of the observed start time\n",
+    "- **actual_range_end**: an ISO 8601 formatted string of the observed end time\n",
+    "- **message**: a message to describe why the frequency cannot be inferred\n",
+    "- **estimated_freq**: a message to describe why the frequency cannot be inferred\n",
```

Comment:
since `estimated_freq` is a large part of a user's goal in getting the debug object, it would be nice to add a code cell showing how you get that value from the output of `infer_temporal_frequencies`. You could stick it right before this, noting that the first element of the tuple being `None` indicates that there was not a perfect frequency. Then that leads into digging into what the rest of the elements of the debug dictionary mean.

Reply:
Done